Sources/Sentry/SentrySDK.m

```diff
@@ -83,6 +83,28 @@ + (void)startWithOptionsObject:(SentryOptions *)options
     [SentrySDK installIntegrations];
 }
 
++ (void)startWithDsn:(NSString *)dsn
```

Comment:
H: `dsn` belongs to the options, why do we have a specific field for it?
also, this goes against our https://develop.sentry.dev/sdk/unified-api/#static-api

Reply:
I with Manoel on this one lets not special case DSN, it should be part of the block as well.

Reply:
I always saw the `init("dsn")` as a very useful overload. In simple samples it's the easiest way to show how to use Sentry. For that reason I even had raised an issue to get this to Sentry Android (for Java at least which will use the init programatically more than Android): https://app.asana.com/0/1159797622087522/1178966110073923

Since we aim to have sensible defaults, it's very common that users want to provide the DSN and nothing else to the SDK. That means intantianting Options or passing a callback to set only the DSN to options is very cumbersome. Specially when we want to show the users how simple the SDK is.

That said, I'd vote for having the string overload if y'all agree.

Basic usage of the .NET SDK:

https://github.com/getsentry/sentry-dotnet/blob/d2c16b3f837ce24778f49bc73ec39ab74f9c8470/samples/Sentry.Samples.Console.Basic/Program.cs#L7-L11

Reply:
tbh I'm fine with an overload for DSN (I'd bring this up to the TSC though) and it makes sense to me too, but this would not fix the problem though.

the main reason here is if users want to set their own options and they still need to use a map, not having a compile safety for its types.

Reply:
The users wouldn't need to use a map @marandaneto. They can do this:
```
SentrySDK.start(dsn: "___PUBLIC_DSN___") { options in
    options.beforeSend = { event in
        return event
    }
    options.debug = true
}
```

Reply:
@philipphofmann the problem with this is: you have a field for the DSN and the options itself has a field for DSN, which one wins?
for me its either:

```
SentrySDK.start { options...
}
```

and/or

```
SentrySDK.start(dsn: DSN) // it uses all the default values
```